fix: Exclude test for rising animation state count during runtime builds

NoelStephensUnity opened the pull request:
Since the test project does not have access to the Unity.Netcode.Components internals during runtime, this PR only does this check when the integration tests are run from within the editor.

NoelStephensUnity committed:
fix

only include the GetAnimationMessage when running in the editor.

NoelStephensUnity committed:
test

only include the animation state count check when running in the editor

jeffreyrainy commented on testproject/Assets/Tests/Runtime/Animation/AnimatorTestHelper.cs:
```diff
@@ -29,10 +29,12 @@ private void Awake()
             m_NetworkAnimator = GetComponent<NetworkAnimator>();
         }
 
+#if UNITY_EDITOR
```
I understand the `#if` in `NetworkAnimatorTests.cs`. But why do need this one too? Couldn't we leave this exposed, always, but not called outside the editor ?

NoelStephensUnity replied:
We could, but why keep an internal method for runtime builds that will never be used during runtime and is not part of the SDK?

jeffreyrainy commented on com.unity.netcode.gameobjects/Components/NetworkAnimator.cs:
```diff
@@ -549,6 +549,7 @@ public override void OnDestroy()
         private ClientRpcParams m_ClientRpcParams;
         private AnimationMessage m_AnimationMessage;
 
+#if UNITY_EDITOR
```
I understand the `#if` in `NetworkAnimatorTests.cs`. But why do need this one too? Couldn't we leave this exposed, always, but not called outside the editor ?

NoelStephensUnity replied:
We could, but why keep an internal method for runtime builds that will never be used?

jeffreyrainy replied:
Whether it is called or not is a consideration of the calling code. If we try to #if out everything that can't be called in any specific case, there will be an explosion of #if s all over the code. It's an internal method, it's there. All good. What I worry about is the #if that can lead to highly unmaintainable code

jeffreyrainy replied:
Don't get me started on the time that [large Montreal game company] threw in a `#ifdef fifty_journalists` because it had a journalist delegation coming and needed a demo. It was supposed to be just for a build, for one afternoon. That define stayed there 6+ months, across multiples games and nobody seemed able to remove it once it started affecting the whole codebase.

NoelStephensUnity committed:
update

removing #ifdef from NetworkAnimator.
Adding comments above the two #ifdefs' in test project.
